x-pack/qa/runtime-fields/core-with-search/src/yamlRestTest/java/org/elasticsearch/xpack/runtimefields/test/search/CoreTestsWithSearchRuntimeFieldsIT.java

```diff
@@ -220,7 +220,7 @@ protected boolean handleIndex(IndexRequest index) {
                         }
                         // Strings are funny, the regular dynamic mapping puts them in "name.keyword" so we follow along.
                         Map<String, Object> keyword = new HashMap<>(runtimeFieldLoadingFromSource("keyword"));
-                        keyword.put("script", "emit(params._source." + name + ");");
+                        keyword.put("script", "emit(params._source['" + name + "']);");
```

Comment:
If it has an issue resolving nested fields now, you could add an if condition so that the script either emits `params._source." + name.replace('.', '.?')` or `params._source['" + name + "']`, depending on what's non-null.

Reply:
I don't think replacing things with `.?` will work here, as internally the values will be a flat map with dots in the keys, not a series of nested maps.

Reply:
It seems to work well in its current form for both dotted and nested fields, as the test still covers the entire suite and it passes.